plugins/disabled-ZeronameLocal/SiteManagerPlugin.py

```diff
@@ -0,0 +1,153 @@
+import logging, json, os, re, sys, time
+from Plugin import PluginManager
+from Config import config
+from Debug import Debug
+from http.client import HTTPSConnection, HTTPConnection
+from base64 import b64encode
+
+allow_reload = False # No reload supported
+
+log = logging.getLogger("ZeronameLocalPlugin")
+
+
+@PluginManager.registerTo("SiteManager")
+class SiteManagerPlugin(object):
+
+    def load(self, *args, **kwargs):
+        super(SiteManagerPlugin, self).load(*args, **kwargs)
+        url = "%(host)s:%(port)s" % {"host": config.namecoin_host, "port": config.namecoin_rpcport}
+        self.c = HTTPConnection(url)
+        user_pass = "%(user)s:%(password)s" % {"user": config.namecoin_rpcuser, "password": config.namecoin_rpcpassword}
+        userAndPass = b64encode(bytes(user_pass, "utf-8")).decode("ascii")
+        self.headers = {"Authorization" : "Basic %s" %  userAndPass, "Content-Type": " application/json " }
+        self.cache = dict()
+
+    # Checks if it's a valid address
+    def isAddress(self, address):
+        return self.isBitDomain(address) or super(SiteManagerPlugin, self).isAddress(address)
+
+    # Return: True if the address is domain
+    def isDomain(self, address):
+        return self.isBitDomain(address) or super(SiteManagerPlugin, self).isDomain(address)
+
+    # Return: True if the address is .bit domain
+    def isBitDomain(self, address):
+        return re.match(r"(.*?)([A-Za-z0-9_-]+\.bit)$", address)
```

Comment:
This is the code that was already in the Zeroname plugin. I believe namecoin only allows you to use numbers, letters and - 

You can also have subdomain like 'me.lola.bit' and it would be valid.

Reply:
Actually namecoin does support utf8 domain names, but I think it's better avoid it because of domain name squatting